enterprise/api/v1/serializers.py

```diff
@@ -606,6 +648,21 @@ def validate(self, data):  # pylint: disable=arguments-differ
             if 'email' in data and not bool(data['email']):
                 raise serializers.ValidationError({'email': ['This field is required']})
 
+        # validate that enterprise customer catalogs exist in the system.
+        enterprise_catalogs = data.get('enterprise_catalogs')
+        enterprise_customer = data['enterprise_customer']
+        if enterprise_catalogs:
+            catalog_count = enterprise_customer.enterprise_customer_catalogs.filter(
+                uuid__in=enterprise_catalogs
+            ).count()
+
+            if catalog_count != len(enterprise_catalogs):
+                raise serializers.ValidationError({
+                    'enterprise_catalogs': [
+                        'You can only link catalogs belonging to the provided enterprise customer.',
```

Comment:
nit: "Only those catalogs can be linked that belong to the enterprise customer. "